src/main/java/core/mate/academy/service/MachineProducer.java

```diff
@@ -2,10 +2,10 @@
 
 import java.util.List;
 
-public interface MachineProducer {
+public interface MachineProducer<T> {
```

Comment:
The MachineProducer interface should be parameterized to restrict its type to Machine or its subtypes to prevent non-Machine types from being used.